src/components/tl/mlx5/tl_mlx5_team.c

```diff
@@ -169,40 +146,153 @@ static inline ucc_status_t ucc_tl_mlx5_a2a_team_test(ucc_base_team_t *team)
+            if (tl_team->global_status_array[UCC_TL_MLX5_MCAST_STATUS_INDEX] != UCC_OK) {
+                /* mcast context is not available for some of the team members so we cannot create
+                 * mcast team */
+                tl_debug(UCC_TL_TEAM_LIB(tl_team),
+                         "failure during mcast ctx create, no mcast team support");
+
+                if (tl_team->local_mcast_ctx_ready) {
+                    comm = tl_team->mcast->mcast_comm;
+                    /* release the resources */
+                    if (ibv_dereg_mr(comm->grh_mr)) {
+                        tl_warn(UCC_TL_TEAM_LIB(tl_team),
+                                "ibv_dereg_mr failed");
+                    }
+                    if (ibv_destroy_cq(comm->rcq)) {
+                        tl_warn(UCC_TL_TEAM_LIB(tl_team),
+                                "ibv_destroy_cq failed");
+                    }
+                    ucc_free(comm->params.oob);
+                    ucc_free(comm);
+                    ucc_free(tl_team->mcast);
+                }
+                tl_team->mcast_state = TL_MLX5_TEAM_STATE_MCAST_NOT_AVAILABLE;
+            } else {
+                tl_debug(UCC_TL_TEAM_LIB(tl_team),
+                         "all team members have mcast ctx ready");
+                tl_team->mcast_state = TL_MLX5_TEAM_STATE_MCAST_INIT;
+            }
+
+            return UCC_INPROGRESS;
+        } else {
+            if (tl_team->global_status_array[UCC_TL_MLX5_A2A_STATUS_INDEX] != UCC_OK) {
+                //a2a team not avail for some of nodes so disable it
+                if (tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_READY) {
+                    // free the resources
+                    ucc_tl_mlx5_alltoall_cleanup(tl_team);
+                }
+                tl_team->a2a_state = TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE;
+            }
+
+            if (tl_team->global_status_array[UCC_TL_MLX5_MCAST_STATUS_INDEX] != UCC_OK) {
+                //mcast team not avail for some of nodes so disable it
+                if (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_READY) {
+                    // free the resources
+                    ucc_tl_mlx5_clean_mcast_comm(tl_team->mcast->mcast_comm);
+                }
+                tl_team->mcast_state = TL_MLX5_TEAM_STATE_MCAST_NOT_AVAILABLE;
+            }
+
+            tl_debug(team->context->lib, "attempted to initialize tl team: %p: MCAST component is %s ALLTOALL component is %s",
+                    team, (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_READY)?"ENABLED":"DISABLED",
+                    (tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_READY)?"ENABLED":"DISABLED");
+        }
+
+        if (tl_team->mcast_state == TL_MLX5_TEAM_STATE_MCAST_NOT_AVAILABLE &&
+            tl_team->a2a_state == TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE) {
+            tl_warn(team->context->lib, "unable to initialize tl team as both a2a and mcast are not available: %p", team);
```

Comment:
please make log messages style identical, sometime it's ALLTOALL or alltoall or a2a